### PR TITLE
Fix bug to not print both long chars

### PR DESCRIPTION
**Pull request by gogo2464**
**Checklist**

- [ ] Closing issues: #issue
- [ ] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [radare2book](https://github.com/radareorg/radare2book)

**Description**

Fix the bug that write 2 times multiple charsets.

### CODE REVIEW

File: libr/util/charset.c [hunk 1]
```diff
@@ -149,6 +149,7 @@ R_API size_t r_charset_decode_str(RCharset *rc, ut8 *out, size_t out_len, const
 					o += ll - 1;
 				}
 				found = true;
+				cur += strlen (str) - 1;
```
Comment (trufae) [inline libr/util/charset.c:152]: Strlen(str) is the same as j

Comment (gogo2464) [reply libr/util/charset.c:152]: No sometimes they do not have the same value.

Comment (trufae) [inline libr/util/charset.c:152]:
```suggestion
				size_t str_len = strlen (str) - 1;
				cur += str_len > 0 ? str_len - 1: 1;
```

**Review by trufae — APPROVED** [review]

**Comment by trufae** [comment]
I just fixed the possible underflow and ill merge, but this strlen is unnecessary to me, we can optimize it later. lets just move forward. sorry for the slow reposnse, busy in the real world lately

**Pull request merged**